### ISSUE TITLE
phx.gen.auth not showing diff between 1.7.0-rcs and 1.7.0 release

### ISSUE THREAD

**Issue by doughsay** [zestcreative/utility#22]
Phoenix 1.7.0 final release has differences in how the simple_form component is used and it affects the generated auth liveviews.  Here's a generated diff: https://elixirstream.dev/gendiff/phx_new/C1AD861F69C1694392C48F3C2179C82A

For example, it doesn't show any diffs for `user_login_live.ex` when it should. Example mount function difference:

1.7.0-rc.1:

```elixir
def mount(_params, _session, socket) do
  email = live_flash(socket.assigns.flash, :email)
  {:ok, assign(socket, email: email), temporary_assigns: [email: nil]}
end
```

1.7.0:

```elixir
def mount(_params, _session, socket) do
  email = live_flash(socket.assigns.flash, :email)
  form = to_form(%{email: email}, as: "user")
  {:ok, assign(socket, form: form), temporary_assigns: [form: form]}
end
```

**Comment by dbernheisel** [comment]
yep, looks like `phx.gen.auth` now interactively prompts now for --live or --no-live if not specified, so the builder fails to generate the auth stuff. Thanks for the report.

**Comment by dbernheisel** [comment]
I will be clearing the diff cache for those created after the phoenix RCs released (Nov 7, 2022 onward)

**Comment by dbernheisel** [comment]
On elixirstream.dev, it'll now require to specify `--no-live` or `--live` for phx.gen.auth for phoenix 1.17.0-rc0 and above which mimics how `mix phx.gen.auth ...` will prompt the user; so there is no way to leave that flag unanswered since the generator will not assume one way or the other.

FWIW, I tested manually with 1.7.0-rc0 and checking user_login_live.ex, and the missing `to_form` line in your example is indeed there. I don't see much auth-related changes between rc0 and final. Please check again; I could be wrong.

https://elixirstream.dev/gendiff/phx_new/639AD8C21FD8CB22B72797EE5597118A

edit: the diff was regenerated above with the fix, so now it shows the line mentioned in the issue.

**Comment by doughsay** [comment]
Hmmm... no something still seems off.  I tried regenerating my specific diff, and I checked the link you posted and both of them are still missing any mentions of any of the generated auth live views and forms.  There should definitely be (as far as I'm aware) a change between rc.0/rc.1 and final in how the live auth views and templates are generated.  Here's what I just now generated: https://elixirstream.dev/gendiff/phx_new/9ECFBA4EEDA4315D12C41B2383EF2889

I definitely specified `--live`, so it shouldn't have gotten hung up on the prompt...  Also I navigated away from the output without reading it 😅 so I missed any details it may have shown me...

**Comment by doughsay** [comment]
Sorry, I think I misunderstood.  You're saying that generating a fresh app with the rc.0 generators added the missing line.  I just tested this myself and am seeing this too, which was definitely not expected, so I dug deeper.  Here's the actual problem:  using `phx_new` version 1.7.0-rc.0 actually results in generating a phoenix project with 1.7.0 final in its deps 😅 

Even though the `mix.exs` looks like this:

```elixir
defp deps do
  [
    {:phoenix, "~> 1.7.0-rc.0", override: true},
    ...
  ]
end
```

The `mix.lock` file looks like this:

```elixir
%{
  ...
  "phoenix": {:hex, :phoenix, "1.7.0", "cbed113bdc203e2ced75859011fe7e71eeebb6259cefa54de810d9c7048b5e22", [:mix], [{:castore, ">= 0.0.0", [hex: :castore, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}, {:phoenix_pubsub, "~> 2.1", [hex: :phoenix_pubsub, repo: "hexpm", optional: false]}, {:phoenix_template, "~> 1.0", [hex: :phoenix_template, repo: "hexpm", optional: false]}, {:phoenix_view, "~> 2.0", [hex: :phoenix_view, repo: "hexpm", optional: true]}, {:plug, "~> 1.14", [hex: :plug, repo: "hexpm", optional: false]}, {:plug_cowboy, "~> 2.6", [hex: :plug_cowboy, repo: "hexpm", optional: true]}, {:plug_crypto, "~> 1.2", [hex: :plug_crypto, repo: "hexpm", optional: false]}, {:telemetry, "~> 0.4 or ~> 1.0", [hex: :telemetry, repo: "hexpm", optional: false]}, {:websock_adapter, "~> 0.4", [hex: :websock_adapter, repo: "hexpm", optional: false]}], "hexpm", "8526139d4bd79ec97c5c3c8e69f6cd663597f782756cec874ba7da5429c93e34"},
  ...
}
```

So while the initial project generation will include the old generator output, the subsequent running of the `phx.gen.auth` generators is actually using the final 1.7.0 release.  Therefore we never actually see the diff we expect.

😭

**Comment by dbernheisel** [comment]
Good find! I'll check it out tonight. Might just remove the `~>` so it's exact.

**Comment by dbernheisel** [comment]
thanks so much @doughsay for revealing that issue. I have a solution locally so I'll get fixed up and released tonight.

**Comment by dbernheisel** [comment]
Resolved again with 3591525e0029afa35d0e1fd6927a9f6ce6fad8ef
